File: app/src/main/java/com/example/simplehabitassignment/data/vos/models/CategoriesAndProgramsModel.java
package com.example.simplehabitassignment.data.vos.models;



import com.example.simplehabitassignment.data.vos.CategoriesAndProgramsVO;

import java.util.List;

public interface CategoriesAndProgramsModel {



    void getCategoriesAndPrograms(String accessToken, int page, CategoriesAndProgramsDelegate delegate);

    interface  CategoriesAndProgramsDelegate
     {
         void onSuccess(List<CategoriesAndProgramsVO> categoriesProgramsList);

         void onFail(String errorMessage);



     }
}

File: app/src/main/java/com/example/simplehabitassignment/data/vos/models/CategoriesAndProgramsModel.kt
package com.example.simplehabitassignment.data.vos.models


import com.example.simplehabitassignment.data.vos.CategoriesAndProgramsVO

interface CategoriesAndProgramsModel {


    fun getCategoriesAndPrograms(accessToken: String, page: Int, delegate: CategoriesAndProgramsDelegate)

    interface CategoriesAndProgramsDelegate {
        fun onSuccess(categoriesProgramsList: List<CategoriesAndProgramsVO>)

        fun onFail(errorMessage: String)


    }
}
